plugins/webp-uploads/helper.php

```diff
@@ -468,3 +468,27 @@ function webp_uploads_get_mime_type_image( int $attachment_id, string $src, stri
 
 	return null;
 }
+
+/**
+ * Retrieves the MIME type of a file, checking the file directly if possible,
+ * and falling back to the attachment's MIME type if needed.
+ *
+ * The function attempts to determine the MIME type directly from the file.
+ * If that information is unavailable, it uses the MIME type from the attachment metadata.
+ * If neither is available, it defaults to an empty string.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $file          The path to the file.
+ * @param int    $attachment_id The attachment ID.
+ * @return string The MIME type of the file, or an empty string if not found.
+ */
+function webp_uploads_get_file_mime_type( string $file, int $attachment_id ): string {
```

Comment:
Maybe rename this to `webp_uploads_get_attachment_file_mime_type()`? This clarifies the function is intended for attachments, not _any_ files.

Another idea: Maybe make `$attachment_id` the first parameter, and then `$file` as second parameter could be optional? If it is not passed, we could get the file ourselves by calling `get_attached_file()` in the function, so that would be a good default.

It would allow us to simplify code in places where we need to get the file only for this check, while still being efficient for other places where we need to use the file for other purposes as well.

Reply:
Thanks for the feedback. Make sense. Open #1662 for review and merge as it's quick one.